### PR TITLE
localCI: improve message errors when the git command fails

### LINKED ISSUE #184: localCI: improve git message errors
when the git command fails the message error is quite simple and it makes sometimes hard to debug the issue

### PR DESCRIPTION
**Pull request by devimc**
this patch includes git's stderr in the error returned by
downloadPullRequest

fixes #184

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

File: cmd/localCI/github.go [hunk 1]
```diff
@@ -208,16 +209,23 @@ func (g *Github) downloadPullRequest(pr int, workingDirectory string) (string, e
...
 
+	var stderr bytes.Buffer
+
 	cmd := exec.Command("git", "clone", g.url, ".")
 	cmd.Dir = projectDirectory
+	cmd.Stderr = &stderr
 	if err := cmd.Run(); err != nil {
-		return "", fmt.Errorf("failed to clone project %s", err)
+		ciLog.Errorf("%s", stderr.String())
+		return "", fmt.Errorf("failed to run git clone %s", err)
```
Comment (jodh-intel) [inline cmd/localCI/github.go]: How about:

```
return "", fmt.Errorf("failed to run %+v: %s", cmd, err)
```

File: cmd/localCI/github.go [hunk 2]
```diff
@@ -208,16 +209,23 @@ func (g *Github) downloadPullRequest(pr int, workingDirectory string) (string, e
...
 	}
 
+	stderr.Reset()
 	cmd = exec.Command("git", "pull", "--no-edit", "origin", fmt.Sprintf("pull/%d/head", pr))
 	cmd.Dir = projectDirectory
+	cmd.Stderr = &stderr
 	if err := cmd.Run(); err != nil {
-		return "", fmt.Errorf("failed to pull request %s", err)
+		ciLog.Errorf("%s", stderr.String())
+		return "", fmt.Errorf("failed to run git pull %s", err)
```
Comment (jodh-intel) [inline cmd/localCI/github.go]: Same comment as above.

Comment (devimc) [inline cmd/localCI/github.go]: it was my first option, unfortunately sometimes the error message is big, I prefer to log it and return an error

**Comment by jodh-intel** [comment]
lgtm

**Comment by grahamwhaley** [comment]
lgtm

**Comment by grahamwhaley** [comment]
@jodh-intel needs your review signoff

**Comment by jodh-intel** [comment]
Looks like pullapprove is confused. Let me try a quick label toggle...

**Comment by jodh-intel** [comment]
Yep - that did it :)

**Pull request merged**